RocketChat/Rocket.Chat
# [FIX] Several issues related to custom roles

## Proposed changes (including videos or screenshots)


- Throw an error when trying to delete a role (User or Subscription role) that are still being used;
- Fix "Invalid Role" error for custom roles in Role Editing sidebar;
- Fix "Users in Role" screen for custom roles.



## Issue(s)

Fix #23989
Close #24059 

## Steps to test or reproduce

1 Go to **Admin > Permissions > +** and create a new Global (scope) role;
2. Select the new role in the list of roles (table header). The "Role Editing" sidebar will be shown. Go to **"Users in Role"** and assign users to this role;
3. Return to the Permissions page, select the same role and try to delete it. The following **error** should be displayed: `Cannot delete role because it's in use [error-role-in-use]`;
4. Repeat this process for a role in the "Rooms" scope.
5. Try to delete roles from users in the "Users in role" page. No error should be shown for both scopes.

https://user-images.githubusercontent.com/36537004/149013092-c305f183-3b56-4932-aaa3-434fed7d2ddf.mp4


https://user-images.githubusercontent.com/36537004/149020300-474e02c3-b371-40e2-acaf-b1d4aa7bbe5c.mp4


## Further comments
